src/realm/obj.cpp

```diff
@@ -1935,6 +1935,91 @@ bool Obj::remove_backlink(ColKey col_key, ObjLink old_link, CascadeState& state)
+                // Single link
+                REALM_ASSERT(!linking_obj.get<ObjKey>(c) || linking_obj.get<ObjKey>(c) == other.get_key());
+                linking_obj.set(c, get_key());
+            }
+            else {
+                auto l = linking_obj.get_linklist(c);
+                auto n = l.find_first(other.get_key());
+                REALM_ASSERT(n != realm::npos);
+                l.set(n, get_key());
+            }
+        }
+        return false;
+    };
+    m_table->for_each_backlink_column(copy_links);
+}
+
+void Obj::dup_and_handle_multiple_backlinks()
+{
+    REALM_ASSERT(!m_table->get_primary_key_column());
+
+    auto copy_links = [this](ColKey col) {
+        auto t = m_table->get_opposite_table(col);
+        auto c = m_table->get_opposite_column(col);
+        auto backlinks = get_all_backlinks(col);
+        for (auto bl : backlinks) {
+            auto obj = m_table->create_object();
+            obj.copy_other_object(*this);
+
+            auto linking_obj = t->get_object(bl);
+
+            // the central piece that is missing is how we handle if there are multiple
+            // embedded objects that are parent of the one just created.
+
+            // backlink from list
+            if (c.get_type() == col_type_LinkList) {
+                auto linking_obj_list = linking_obj.get_linklist(c);
+                auto n = linking_obj_list.find_first(get_key());
+                REALM_ASSERT(n != realm::npos);
+                linking_obj_list.set(n, obj.get_key());
+            }
+            // backlink from set
+            else if (c.get_attrs().test(col_attr_Set)) {
+                // this SHOULD NEVER HAPPEN, since SET sematincs forbids to have a backlink to the same object store
+                // multiple times. update_schema should throw longer before to perform this copy.
+                auto linking_obj_set = linking_obj.get_setbase_ptr(c);
+                auto pos = linking_obj_set->find_any(Mixed{get_key()});
+                REALM_ASSERT(pos != realm::npos);
+                linking_obj_set->insert_any(obj.get_key());
+            }
+            // backlink from dictionary
+            else if (c.get_attrs().test(col_attr_Dictionary)) {
+                auto linking_obj_dictionary = linking_obj.get_dictionary_ptr(c);
+                auto pos = linking_obj_dictionary->find_any(ObjLink{m_table->get_key(), get_key()});
+                REALM_ASSERT(pos != realm::npos);
+                Mixed key = linking_obj_dictionary->get_key(pos);
+                linking_obj_dictionary->insert(key, obj.get_key());
+            }
+            // backlink from mixed
+            else if (c.get_type() == col_type_Mixed && linking_obj.get_any(c).get_type() == type_TypedLink) {
+                // this is not supported yet.
+                REALM_ASSERT(!linking_obj.get<ObjKey>(c) || linking_obj.get<ObjKey>(c) == get_key());
+                linking_obj.set_any(c, obj.get_key());
```

Comment:
This needs to set the value as a `ObjLink` not a `ObjKey`, otherwise there is no destination table.

Reply:
Yes, I actually did for searching but not for inserting… my bad